src/identity_policy_provider.py

```diff
@@ -0,0 +1,208 @@
+            "type": "string",
+            "description": "that the policy will apply to"
+        },
+        "PolicyName": {
+            "type": "string",
+            "maxLength": 64,
+            "pattern": "^[A-Za-z-_]+$",
+            "description": "of the policy"
+        },
+        "PolicyDocument": {
+            "type": "object",
+            "description": "the permissions for the principal",
+            "required": ["Statement"],
+            "properties": {
+                "Statement": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "required": ["Effect", "Principal", "Action"],
+                        "properties": {
+                            "Effect": {
+                                "type": "string",
+                                "enum": ["Allow", "Deny"]
+                            },
+                            "Principal": {
+                                "type": "object",
+                                "oneOf": [{
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "string"
+                                        }
+                                    }
+                                }, {
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "array",
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        }
+                                    }
+                                }]
+                            },
+                            "Action": {
+                                "type": "array",
+                                "items": {
+                                    "type": "string",
+                                    "enum": ["ses:SendEmail", "ses:SendRawEmail"]
+                                }
+                            }
+                        }
+                    }
+                },
+                "Version": {
+                    "type": "string",
+                    "description": "of the policy document"
+                }
+            }
+        }
+    }
+}
+
+
+class IdentityPolicyProvider(ResourceProvider):
+
+    def __init__(self):
+        super(IdentityPolicyProvider, self).__init__()
+        self.request_schema = request_schema
+        self.ses = boto3.client('ses')
+
+    def convert_property_types(self):
+        self.heuristic_convert_property_types(self.properties)
+
+    def create(self):
+        existing_policy = self.get_policy(self.get('Identity'), self.get('PolicyName'))
```

Comment:
In this code you see a lot of references to the identity and policy name. I often add a utility method:

```python
@property
def policy_name(self):
    return self.get('PolicyName')

@property
def old_policy_name(self):
    return self.get_old('PolicyName', self.policy_name)

@property
def identity(self):
    return self.get('Identity')

@property
def old_identity(self):
    return self.get_old('Identity', self.identity)

```

So that the code becomes slightly kinder to the eye..